ee/clickhouse/models/cohort.py

```diff
@@ -68,19 +68,33 @@ def format_person_query(cohort: Cohort, index: int, **kwargs) -> Tuple[str, Dict
     return joined_filter, params
 
 
-def get_properties_cohort_subquery(cohort: Cohort, cohort_group: Dict, group_idx: int):
+def get_properties_cohort_subquery(cohort: Cohort, cohort_group: Dict, group_idx: int) -> Tuple[str, Dict[str, Any]]:
     from ee.clickhouse.models.property import prop_filter_json_extract
 
     filter = Filter(data=cohort_group)
     params: Dict[str, Any] = {}
 
     query = ""
     for idx, prop in enumerate(filter.properties):
-        filter_query, filter_params = prop_filter_json_extract(
-            prop=prop, idx=idx, prepend="{}_{}_{}_person".format(cohort.pk, group_idx, idx)
-        )
-        params = {**params, **filter_params}
-        query += filter_query
+        if prop.type == "cohort":
+            try:
+                prop_cohort: Cohort = Cohort.objects.get(pk=prop.value, team_id=cohort.team_id)
+            except Cohort.DoesNotExist:
+                return "3 = 4", {}
+            if prop_cohort.pk == cohort.pk:
+                # If we've encountered a cyclic dependency (meaning this cohort depends on this cohort),
+                # we treat it as satisfied for all persons
+                query += "AND 7 = 7"
+            else:
+                person_id_query, cohort_filter_params = format_filter_query(prop_cohort, idx, "person_id")
+                params.update(cohort_filter_params)
+                query += f"AND person_max.id IN ({person_id_query})"
```

Comment:
Use `person.id` here instead. `person_max` is an internal implementation detail of `GET_LATEST_PERSON_SQL`